packages/mobile/src/components/drawer/NativeDrawer.tsx

```diff
@@ -17,14 +18,20 @@ type NativeDrawerProps = SetOptional<DrawerProps, 'isOpen' | 'onClose'> & {
  * opening and closing.
  */
 export const NativeDrawer = (props: NativeDrawerProps) => {
-  const { drawerName, onClose: onCloseProp, ...other } = props
+  const {
+    blockClose = true,
+    drawerName,
+    onClose: onCloseProp,
+    ...other
+  } = props
 
   const { isOpen, onClose, onClosed, visibleState } = useDrawer(drawerName)
 
   const handleClose = useCallback(() => {
+    if (blockClose) return
```

Comment:
nit: prefer non-early return here imo

Reply:
Oh interesting I usually feel the other way! What's the reasoning behind wrapping it all in a conditional?

Reply:
at a glance you know the two other methods are fired conditionally, vs having to have the whole fn context. I go either way based on the context - here I think the method is short enough that it makes sense to just use the conditional block. I tend to use early returns in arg validation scenarios and when the block is super large